modules/tappxBidAdapter.js

```diff
@@ -0,0 +1,284 @@
+      publisher.domain = getParameters(validBidRequests, 'params.app.publisher.domain');
+      tagid = `${app.name}_typeAdBanVid_${getOs()}`;
+      payload.app = app;
+      api[0] = getParameters(validBidRequests, 'params.api') ? getParameters(validBidRequests, 'params.api') : [3, 5];
+    } else {
+      let site = {};
+      site.name = window.location.hostname;
+      site.bundle = window.location.hostname;
+      site.domain = window.location.hostname;
+      publisher.name = window.location.hostname;
+      publisher.domain = window.location.hostname;
+      tagid = `${site.name}_typeAdBanVid_${getOs()}`;
+      payload.site = site;
+    }
+    // < App/Site object
+
+    // > Imp object
+    let imp = {};
+    let w;
+    let h;
+
+    if (bannerMediaType) {
+      let banner = {};
+      w = bannerMediaType.sizes[0][0];
+      h = bannerMediaType.sizes[0][1];
+      banner.w = w;
+      banner.h = h;
+      if ((bannerMediaType.sizes[0].indexOf(480) >= 0) && (bannerMediaType.sizes[0].indexOf(320) >= 0)) {
+        banner.pos = 7
+      } else {
+        banner.pos = 4
+      }
+
+      banner.api = api;
+
+      let format = {};
+      format[0] = {};
+      format[0].w = w;
+      format[0].h = h;
+      banner.format = format;
+
+      imp.banner = banner;
+    }
+
+    imp.id = 1;
+    imp.tagid = tagid;
+    imp.secure = 1;
+
+    imp.bidfloor = getParameters(validBidRequests, 'params.bidfloor');
+    // < Imp object
+
+    // > Device object
+    let device = {};
+    // Mandatory
+    device.os = getOs();
+    device.ip = 'peer';
+    device.ua = navigator.userAgent;
+    device.ifa = validBidRequests.ifa;
+
+    // Optional
+    device.h = screen.height;
+    device.w = screen.width;
+    device.dnt = utils.getDNT() ? 1 : 0;
+    device.language = getLanguage();
+    device.make = navigator.vendor ? navigator.vendor : '';
+
+    let geo = {};
+    geo.country = getParameters(validBidRequests, 'params.geo.country');
+    // < Device object
+
+    // > Params
+    let params = {};
+    params.host = 'tappx.com';
+    params.tappxkey = TAPPXKEY;
+    params.endpoint = ENDPOINT;
+    params.bidfloor = BIDFLOOR;
+    // < Params
+
+    // > GDPR
+    let regs = {};
+    regs.gdpr = 0;
+    if (!(bidderRequest.gdprConsent == null)) {
+      if (typeof bidderRequest.gdprConsent.gdprApplies === 'boolean') { regs.gdpr = bidderRequest.gdprConsent.gdprApplies; }
+      if (regs.gdpr) { regs.consent = bidderRequest.gdprConsent.consentString; }
+    }
+
+    // CCPA
+    regs.ext = {};
+    if (!(bidderRequest.uspConsent == null)) {
+      regs.ext.us_privacy = bidderRequest.uspConsent;
+    }
+
+    // COPPA compliance
+    if (config.getConfig('coppa') === true) {
+      regs.coppa = config.getConfig('coppa') === true ? 1 : 0;
+    }
+    // < GDPR
+
+    // > Payload
+    payload.id = (getParameters(validBidRequests, 'auctionId'));
+    payload.test = 1;
+    payload.at = 1;
+    payload.tmax = 500;
```

Comment:
why is this hardcoded, maybe you wanted to read it from `bidderRequest.timeout` ?

Reply:
We have updated the adapter to get the value from biderRequest.tiemout or to set it to 600 if it doesn’t come there. Commit: [94e2357](https://github.com/prebidtappx/Prebid.js/commit/94e235771d29a09862bd52a776b328e350bc35db)